Dockerfile

```diff
@@ -0,0 +1,37 @@
+# sources:
+# - https://github.com/jupyter/docker-stacks
+# - https://github.com/TomographicImaging/CIL#installation-of-cil
+# consumers:
+# - https://github.com/stfc/cloud-docker-images
+# TODO: use `ubuntu-22.04` after `python=3.11` is supported, vis.
+# https://github.com/TomographicImaging/CIL/issues/1490
+FROM jupyter/tensorflow-notebook:ubuntu-20.04
+LABEL org.opencontainers.image.source=https://github.com/TomographicImaging/CIL
+LABEL org.opencontainers.image.description="Core Imaging Library"
+LABEL org.opencontainers.image.licenses="Apache-2.0"
+
+# CUDA-specific packages
+ARG CIL_EXTRA_PACKAGES=astra-toolbox tigre
+# build & runtime dependencies
+# TODO: sync with scripts/requirements-test.yml
+# TODO: sync with scripts/create_local_env_for_cil_development.sh
+RUN mamba install -y -c conda-forge -c intel -c ccpi \
+  $CIL_EXTRA_PACKAGES \
+  _openmp_mutex ccpi-regulariser cmake dxchange h5py libgcc-ng libstdcxx-ng numba numpy pillow pip pywavelets setuptools tomophantom tqdm \
+  python-wget scikit-image packaging \
```

Comment:
any pins required here @gfardell?

related to https://github.com/TomographicImaging/CIL/pull/1590#discussion_r1432818775 🙃

Reply:
I suppose the ideal thing is now to clean up the `scripts` directory:
- getting rid of the bash scripts and 
- adding versions to the yaml files: py39 and np121 and [these](https://github.com/TomographicImaging/CIL/blob/0e644bce2e88c586995c7a266ca281a60f391fa4/recipe/meta.yaml#L94-L98) additional restrictions
- moving the yaml file to the root directory

Reply:
Talking with @gfardell we are changing our mind (constantly). The problem we'd like to solve is that all CI does the same testing without changing the various versions in so many files which is unmaintainable: Jenkins (although this is in a different place), GHA, docker, the install scripts.

How do we do it?

Reply:
while trying `scripts/requirements-test.yml` I ran into https://stackoverflow.com/q/58555389. Will push a work-around here. Will leave cleanup for #1590 (TL;DR looks like we'll need `scripts/create_local_env_for_cil_development.sh` after all)